### ISSUE TITLE
Connect to Oracle 18c/19c Autonomous databases through wallets

### ISSUE THREAD

**Issue by xyu-r0nin-** [dbeaver/dbeaver#6912]
**Is your feature request related to a problem? Please describe.**
DBeaver does not currently support the creation of connections to 18c / 19c databases using Oracle wallets

**Describe the solution you'd like**
It would be great if DBeaver allows you to create connections to Oracle 18c and 19c databases using wallets or jks files:

[Java Connectivity with Autonomous Database (ATP or ADW) using 19c and 18.3 JDBC](https://www.oracle.com/technetwork/database/application-development/jdbc/documentation/atp-5073445.html)

**Describe alternatives you've considered**
Implement the use of wallets or jks files

**Additional context**
This is how the connection is configured in SQL Developer 19.2:
![imagen](https://user-images.githubusercontent.com/7806764/66169091-d50a7200-e604-11e9-947f-5da5bf849c2a.png)

This is the content of Wallet.zip
![imagen](https://user-images.githubusercontent.com/7806764/66169203-3df1ea00-e605-11e9-87b7-88da8dcebdbb.png)

**Comment by jorgefloresesc** [comment]
@glima34  how do you achieved that? I'm still trying to figure out how to connect with dbeaver using wallet. I'm able to connect from toad but not form dbeaver

**Comment by xyu-r0nin-** [comment]
> @glima34 how do you achieved that? I'm still trying to figure out how to connect with dbeaver using wallet. I'm able to connect from toad but not form dbeaver

![imagen](https://user-images.githubusercontent.com/7806764/89714097-e5e0ee00-d961-11ea-8645-a3f98d5631d8.png)

You just have to unzip the wallet and in "Connection settings" select "Custom" and write the url like this:

jdbc:oracle:thin:@<service>?TNS_ADMIN=<path_to_unzipped_wallet>

For example: jdbc:oracle:thin:@myservice_tp?TNS_ADMIN=/app/config/wallets/WalletCloud

**Comment by jorgefloresesc** [comment]
@xyu-r0nin- hi, thanks for your reply. This is what I'm getting using your example:
![image](https://user-images.githubusercontent.com/69055288/89795196-f49fe000-dadc-11ea-8a74-a99e3c6ecff1.png)

BTW, I'm working on windows.

**Comment by jorgefloresesc** [comment]
@xyu-r0nin- Hi, thanks for your reply. This is what I'm getting using your example:
![image](https://user-images.githubusercontent.com/69055288/89795349-21ec8e00-dadd-11ea-80c8-79ea8d534542.png)

BTW, I'm using windows.

**Comment by xyu-r0nin-** [comment]
@jorgefloresesc Are you sure the name of the service is OCIDWQ01

Open the file tnsnames.ora that is inside the wallet and see what are the names of the services:

![imagen](https://user-images.githubusercontent.com/7806764/90015185-e03f1d00-dc6d-11ea-86b3-22fbc2224058.png)

**Comment by mindhub-jean** [comment]
> @xyu-r0nin- Hi, thanks for your reply. This is what I'm getting using your example:
> ![image](https://user-images.githubusercontent.com/69055288/89795349-21ec8e00-dadd-11ea-80c8-79ea8d534542.png)
> 
> BTW, I'm using windows.

Man, i had the same problem. Make sure you are using driver version 18.3 instead of 12.2.0.1 

thanks @xyu-r0nin- for the solution.. u rock !!

**Comment by xyu-r0nin-** [comment]
> > @xyu-r0nin- Hi, thanks for your reply. This is what I'm getting using your example:
> > ![image](https://user-images.githubusercontent.com/69055288/89795349-21ec8e00-dadd-11ea-80c8-79ea8d534542.png)
> > BTW, I'm using windows.
> 
> Man, i had the same problem. Make sure you are using driver version 18.3 instead of 12.2.0.1
> 
> thanks @xyu-r0nin- for the solution.. u rock !!

Hi @ mindhub-jean, I'm glad it worked for you.

I did several tests and I can confirm that this solution applies for ojdbc from version 18.3 to version 19.6. With 19.7 it doesn't work

**Comment by Dekshith** [comment]
Download jars from [https://download.oracle.com/otn/utilities_drivers/jdbc/183/ojdbc8-full.tar.gz](url)
and add those libraries for connection

![image](https://user-images.githubusercontent.com/36989606/102618120-3e21cb00-4160-11eb-83b4-3cee87cd1684.png)

**Comment by voltes-7777** [comment]
had a difficult time with the above suggestions but at least it showed me that there has to be a way since it worked for others. 
this link about creating a database driver finally helped me 
> [dbform link](https://www.dbform.com/2019/11/27/how-to-connect-dbeaver-to-oracle-autonomous-database-by-jdbc-thin-driver/)  

but it's in Chinese characters so using google translate  

> [dbform link translated](https://translate.google.com/translate?hl=en&sl=auto&tl=en&u=https%3A%2F%2Fwww.dbform.com%2F2019%2F11%2F27%2Fhow-to-connect-dbeaver-to-oracle-autonomous-database-by-jdbc-thin-driver%2F)

thanks to Kamus for writing that

**Comment by lukaseder** [comment]
In case it helps anyone, I followed this tutorial: https://docs.oracle.com/en/cloud/paas/autonomous-database/adbsa/connect-jdbc-thin-wallet.html

And had to manually specify the JDBC drivers and security libraries:

![image](https://user-images.githubusercontent.com/734593/123986663-cb26c600-d9c6-11eb-97ff-f5a60f26ffee.png)

Alternatively, even better, add this dependency (once https://github.com/dbeaver/dbeaver/issues/13048 is resolved):

![image](https://user-images.githubusercontent.com/734593/123992542-ff50b580-d9cb-11eb-90a0-09f72aadf15f.png)

Which will download all the transitive dependencies:

![image](https://user-images.githubusercontent.com/734593/123992579-07105a00-d9cc-11eb-9a05-aae2a8b07e31.png)

**Comment by javierdiegof** [comment]
Facing the same problem as others here. I was able to make it work just for driver versions `19.3.0.0` and `19.6.0.0` (with either artifacts or downloading the folders from the Oracle site). I would expect it to work for any `19.*` version if the database is Oracle 19c, right? (which it is in my case, btw)
I've noticed that, if I let DBeaver download the artifacts itself, then for the two versions that work, DBeaver finds the dependencies of the artifact, whereas for the versions that don't work, DBeaver just downloads the artifact by itself without the dependencies. So for example for artifact version `19.3.0.0`:
![Screenshot from 2021-10-20 18-30-46](https://user-images.githubusercontent.com/16964518/138186458-57d66acf-8c3d-497f-9b95-c494d5b65848.png)
Whereas for `19.7.0.0`:
![Screenshot from 2021-10-20 18-32-51](https://user-images.githubusercontent.com/16964518/138186583-a42c124d-1c1b-46a5-9986-1a675c816714.png)
Could this be the source of the error? And if it is, what could be the reason for it to behave that way?
Thanks to everyone for your help :)

(Additional note: I found out a bit later on this [Oracle article](https://blogs.oracle.com/developers/post/your-own-way%E2%80%8A-%E2%80%8Aoracle-jdbc-drivers-19700-on-maven%C2%A0central)) that version `19.7.0.0` was the first one that did not include all of the dependencies inside the main maven file. There, it is claimed that changing the maven dependency from `ojdbcn` to `ojdbcn-production` should make the code equivalent since that new package contains all of the dependencies in it as it was distributed before).
